internal/endpoints/network.go

```diff
@@ -71,6 +71,12 @@ func (n *Network) Listen() error {
 	return nil
 }
 
+// UpdateTLS updates the TLS configuration of the network listener.
+func (n *Network) UpdateTLS(cert *shared.CertInfo) {
+	n.cert = cert
+	n.listener.(*listeners.FancyTLSListener).Config(cert)
```

Comment:
To prevent running into a panic why not switching to:

```go
c, ok := n.listener.(*listeners.FancyTLSListener)
if ok {
    c.Config(cert)
}
```

Maybe also log an error in case this didn't work as expected?